content/doc/book/using/using-agents.adoc

```diff
@@ -0,0 +1,195 @@
+
+1. In a terminal window run the command: `ssh-keygen -f ~/.ssh/jenkins_agent_key`
+2. Provide a passphrase to use with the key
+3. Confirm the output looks something like this:
++
+[source,bash]
+----
+ubuntu@desktop:~$ ssh-keygen -f ~/.ssh/jenkins_agent_key
+Generating public/private rsa key pair.
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+Your identification has been saved in /home/ubuntu/.ssh/jenkins_agent_key
+Your public key has been saved in /home/ubuntu/.ssh/jenkins_agent_key.pub
+The key fingerprint is:
+SHA256:XqxxjqsLlvDD0ZHm9Y2iR7zC6IbsUlMEHo3ffy8TzGs
+The key's randomart image is:
++---[RSA 3072]----+
+|  o+             |
+| ...o  .         |
+|  .o .+ .        |
+|    o+.+ o o     |
+|  ... o.So* .    |
+|  o+ = +.X=      |
+| o oO + *..+     |
+|. oo.o o .E .    |
+| o... oo.. o     |
++----[SHA256]-----+
+----
+
+==== Create a Jenkins SSH credential
+
+1. Go to your Jenkins dashboard;
+2. Go to `Manage Jenkins` option in main menu and click on credentials button;
++
+image:node/credentials-1.png[credentials menu]
+
+3. select the drop option `Add Credentials` from the global item;
++
+image:node/credentials-2.png[add credentials option]
+
+4. Fill the form:
+* Kind:  SSH Username with private key;
+* id: jenkins
+* description: The jenkins ssh key
+* username: jenkins
+* Private Key: select `Enter directly` and press the Add button to insert your private key from `~/.ssh/jenkins_agent_key`
+* Passphrase: fill your passphrase used to generate the SSH key pair and then press OK
+    image:node/credentials-3.png[credentials filled form]
+
+=== Creating your Docker agent
+
+Here we will use the link:https://github.com/jenkinsci/docker-ssh-agent[docker-ssh-agent image] to create the agent containers.
+
+1. run the command to start your first agent:
++
+[source,bash]
+----
+docker run -d --rm --name=agent1 -p 22:22 \
+-e "JENKINS_AGENT_SSH_PUBKEY=[your-public-key]" \
+jenkins/ssh-agent:alpine
+----
++
+[NOTE]
+====
+* Remember to replace the tag [your-public-key] for your own SSH *public* key.
+* Your public key in this example is: `cat ~/.ssh/jenkins_agent_key.pub`
+====
+2. Now run the following command to update the container environment:
++
+[source,bash]
+----
+docker exec  agent1 sh -c "env | egrep -v "^(HOME=|USER=|MAIL=|LC_ALL=|LS_COLORS=|LANG=|HOSTNAME=|PWD=|TERM=|SHLVL=|LANGUAGE=|_=)" >> /etc/environment"
+----
++
+[NOTE]
+====
+The step 2 is necessary because the image is configured to reject changes to environment variables.
+when the link:https://github.com/jenkinsci/docker-ssh-agent/issues/33[issue #33] is fixed, we can ignore this step.
+====
+3. Now the container `agent1` is running. +
+Hint: the command `docker ps` can be used to check if the container is running as expected.
+
+=== Setup up the agent1 on jenkins.
+
+1. Go to your Jenkins dashboard;
+2. Go to `Manage Jenkins` option in main menu;
+3. Go to `Manage Nodes and clouds` item;
++
+image:node/node-1.png[Manage node menu]
```

Comment:
Limit width of image, allow user to open full size image in another window or tab
```suggestion
image:node/node-1.png[Manage node menu,700]
```